usr/src/uts/i86pc/io/vmm/vmm_sol_dev.c

```diff
@@ -1514,6 +1538,11 @@ vmm_do_vm_destroy_locked(vmm_softc_t *sc, boolean_t clean_zsd)
 	}
 	(void) devfs_clean(pdip, NULL, DV_CLEAN_FORCE);
 
+	if (list_is_empty(&vmm_list)) {
```

Comment:
Have you audited this such that we can be sure that vm_destroy() does *NOT* require the hma reg? That is, if we hit the VMM_DESTROY delayed case, we can be destroying the VM without any hma. Is that safe - if so, please comment as such. If not, consider moving the unregistration into vm_destroy path.